### DISCUSSION TITLE
A simple python based GUI for Whisper

### DISCUSSION THREAD

**Discussion by jjroberts88** [openai/whisper#794] [Show and tell]
Hi everyone,

I made a very basic GUI for whisper using tkinter in Python. 
It allows you to either manually add audio files or 'drag and drop' files to the listbox.
The transcribed text appears in the textbox and is automatically copied to the clipboard.
https://github.com/jjroberts88/transcriber
I'm keen to hear any feedback/suggestions and I will continue to adapt it.

I'm very new to this and in the early stages of learning these skills so apologies in advance for any mistakes.
My background is medicine and my use case for this was to look at how we might reduce our secretary workload burden through automatic transcription.

**Comment by garyvcarter** [comment]
I love the concept! The bad news is I couldn't get it to work. I copied your code and named it transcriber.py then ran it in terminal using "python transcriber.py". It throws an errror on transcriber.py", line 128,  "execution_count": null, NameError: name 'null' is not defined. When I change the null to 0 or None and rerun it nothing happens. I am not much of a programmer but apparently there is no null keyword in Python but it likes None. I assume you got this to work with some magic I have yet to install. I would love to see this work because whisper is awesome. You are definitely on to something here!

**Reply by jjroberts88** [reply]
Hi Gary. Thanks for your message and interest. I will try and replicate your error. Have you tried running it as a pynb jupyter notebook? 

I have made some updates so will add them to the repository when I get a chance.

Let me know how you get on.

**Comment by Gally593** [comment]
Hi jjroberts88,

i was triyng your program and it doesn't work with my pc. But i've fixed it. I copy your code in my python editor ( notepad ++ 😄 ) and add : 
        - import torch
        - swap "change model = whisper.load_model("tiny.en") " 
        by 
        - device = 'cuda' if torch.cuda.is_available() else 'cpu'
           model = whisper.load_model("large").to(device)

The error was : 

![2023-08-29_12h02_55](https://github.com/openai/whisper/assets/129105700/d03ab1cf-9def-42dc-9e7b-3dfdcbba0ade)

thanks for your work.

**Reply by jjroberts88** [reply]
Glad you got it working Gally! Yes I was getting a similar error however it would still run fine on macbook pro.
